### PR TITLE
dbms: max_open_files for server. [#METR-21516]

### PR DESCRIPTION
**Pull request by excitoon**

### CODE REVIEW

**Comment by excitoon** [comment]
That will fix problem with calling ```getrlimit``` on OS X, because it returns junk in ```rlim.rlim_max```.

File: dbms/src/Server/Server.cpp [hunk 1]
```diff
@@ -218,16 +218,12 @@ int Server::main(const std::vector<std::string> & args)
...
-			#else
-			LOG_INFO(log, "Setting nofile limit is not supported on MacOS X. Please increase it manually. See MacOS.md for more information. Current limit is '" << old << "'");
-			throw Poco::Exception("setrlimit is not supported on MacOS X");
-			#endif
-
-			LOG_DEBUG(log, "Set rlimit on number of file descriptors to " << rlim.rlim_cur << " (was " << old << ")");
+			rlim.rlim_cur = config().getUInt("max_open_files", rlim.rlim_max);
+			int rc = setrlimit(RLIMIT_NOFILE, &rlim);
+			if (rc != 0)
+				throw Poco::Exception(std::string("Cannot setrlimit (tried rlim_cur = ") + std::to_string(rlim.rlim_cur) + "): " + strerror(errno) + ". Try to specify max_open_files according to your system limits.");
```
Comment (alexey-milovidov) [inline dbms/src/Server/Server.cpp]: Better to use `throwFromErrno` function.

File: dbms/src/Server/config.xml [hunk 2]
```diff
@@ -32,6 +32,9 @@
 	<!-- Maximum number of concurrent queries. -->
 	<max_concurrent_queries>100</max_concurrent_queries>
 
+	<!-- Set limit of open files (default: maximum). -->
+	<!-- <max_open_files>10240</max_open_files> -->
```
Comment (alexey-milovidov) [inline dbms/src/Server/config.xml]: In fact we have 262144 in `init.d` and `/etc/security`.

Comment (alexey-milovidov) [inline dbms/src/Server/config.xml]: Add `Actual on MacOS X `.

Comment (excitoon) [inline dbms/src/Server/config.xml]: Done.

File: dbms/src/Server/Server.cpp (hunk 1 above)
Comment (excitoon) [inline dbms/src/Server/Server.cpp]: Done.

**Pull request merged**